src/components/Navbar/Navbar.vue

```diff
@@ -0,0 +1,1259 @@
+<script setup>
```

Comment:
This is something I forgot to mention, but we should use typescript in our component files. You can enable it by simply adding `lang="ts"` to the script tag.

With that enabled you can also add a type to the first parameter of `handleDropdownButton`